src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -244,11 +247,20 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
       fetchData(cursors).map { sr =>
         val items = sr.items
         val nextCursor = sr.nextCursor
-        val shouldStopEarly = items.isEmpty || (applyTimestampTermination && items.last.lastUpdatedTime >=
-          terminationTimeStamp)
+        val shouldStopEarly = items.nonEmpty && applyTimestampTermination && items.last.lastUpdatedTime >= terminationTimeStamp
         val next =
-          (nextCursor, shouldStopEarly) match {
-            case (Some(cursor), false) =>
+          (nextCursor, items.isEmpty, shouldStopEarly) match {
+            case (Some(cursor), true, _) =>
+              if (syncCursorSaveCallbackUrl.isDefined && jobId.isDefined && cursorName.isDefined) {
```

Comment:
here it won't be called for backfill case as last page cursor will be empty, we probably want to pass the futureSyncCursor here

Reply:
biting the bullet and splitting/wrapping the cursor types between /sync and /query might've helped to catch that stuff

Reply:
note that /query endpoint may currently return `nextCursor:{}` https://github.com/cognitedata/service-contracts/pull/2468#issuecomment-2017401876 but we'll go and try extract `val nextCursor = qr.nextCursor.getOrElse(Map.empty).get("sync")` "sync" entry and that will be empty on last /query page

Reply:
We don't want to persist the backfill cursor, only those coming from /sync. The backfill ends up giving you that at the end.

Reply:
It doesn't, backfill cursor gets injected on per-item level while cursors remain the same matching the iteration method
https://github.com/cognitedata/cdp-spark-datasource/blob/master/src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala#L230
https://github.com/cognitedata/cdp-spark-datasource/blob/master/src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala#L193

Reply:
So we need to pick up futureSync cursor like for items when in isBackfill mode

Reply:
Indeed, we need to pick up the projected cursor, not the nextCursor.